components/polymer/dust-demo.html

```diff
@@ -75,39 +70,19 @@ <h3 id="output-header" on-click="{{collapseContainer}}">Output</h3>
       :host(:hover) #devToggle {
         opacity: 0.7;
       }
-
-      .container {
-        position: relative;
-      }
-
-      :host /deep/ h3 {
-        position: absolute;
-        top: 0;
-        right: 0;
-        padding: 5px;
-        margin: 0;
-        color: #FFF;
-        border: 1px dashed #72CC72;
-        border-right: none;
-        border-top: none;
-        font-family: Helvetica, sans-serif;
-        font-size: 15px;
-        font-weight: normal;
-      }
     </style>
   </template>
 
   <!-- Interaction -->
   <script>
     Polymer({
-      ready: function() {
-        this.templateName = this.getAttribute('template-name') || 'default' + (new Date()).getTime();
+      attached: function() {
         this.setupListeners();
-        this.renderDust();
         this.onMutation(this, function() {
           this.setupEditorListeners();
           this.renderDust();
         }.bind(this));
+        this.templateName = this.templateName || 'default' + (new Date()).getTime();
```

Comment:
`templateName` will actually still be set before the first call to `renderDust`, through this windy road:

1. `this.templateName = ...` on line 85
2. triggers `templateNameChanged()`
3. which calls `handleEditorUpdate()`
4. which calls `renderDust()`

And then line 83 up there will take care of re-calling renderDust() when the user edits the textarea, I think.